### PR TITLE
Ed card button

### PR DESCRIPTION
**Pull request by Duncid**
## Description
Updating Button

## Risk
Breaking at deploy?
To be tested

### CODE REVIEW

File: sparkle/src/components/CardButton.tsx [hunk 1]
```diff
@@ -56,10 +31,21 @@ const cardButtonVariants = cva(
...
+  rel?: string;
+  replace?: boolean;
+  shallow?: boolean;
+}
+
 export function CardButton({
   children,
-  variant,
-  size,
+  variant = "primary",
```
Comment (JulesBelveze) [inline sparkle/src/components/CardButton.tsx:47]: If you add a "defaultVariants" in cva you do not need to set a default value in the props 😃

File: front/package.json [hunk 2]
```diff
@@ -191,5 +191,6 @@
     "fs": false,
     "dgram": false,
     "child_process": false
-  }
+  },
```
Comment (JulesBelveze) [inline front/package.json:194]: I think you ran `npm version patch` in the wrong folder 😇

File: sparkle/src/components/CardButton.tsx [hunk 3]
```diff
@@ -7,47 +7,22 @@ import {
...
+      variant: {
+        primary: "s-bg-structure-50 s-border-border-dark/0",
+        secondary: "s-bg-background s-border-border-dark",
+        tertiary: "s-bg-background s-border-border-dark/0",
+      },
+      size: {
+        sm: "s-p-3 s-rounded-xl",
+        md: "s-p-4 s-rounded-2xl",
+        lg: "s-p-5 s-rounded-3xl",
+      },
```
Comment (JulesBelveze) [inline sparkle/src/components/CardButton.tsx:25]: Could we keep `sizeVariants` and `variantClasses`?

File: sparkle/src/stories/CardButton.stories.tsx [hunk 4]
```diff
@@ -101,17 +111,17 @@ const cardData: CardData[] = [
...
       <CardButton
         key={index}
-        variant="secondary"
+        variant="primary"
         onClick={() => {
           alert(`You clicked on ${card.title}`);
         }}
       >
-        <div className="s-flex s-w-full s-flex-col s-gap-2 s-text-sm">
+        <div className="s-flex s-w-full s-flex-col s-gap-0 s-text-sm">
```
Comment (JulesBelveze) [inline sparkle/src/stories/CardButton.stories.tsx:124]: Not sure but I think you can remove "s-gap-0"

File: sparkle/src/components/CardButton.tsx [hunk 5]
```diff
@@ -69,15 +55,21 @@ export function CardButton({
...
+
   if (href) {
     return (
       <Link
         href={href}
-        className={cardButtonVariants({ variant, size, className })}
+        className={cardButtonClassNames}
         replace={replace}
         shallow={shallow}
         target={target}
```
Comment (JulesBelveze) [inline sparkle/src/components/CardButton.tsx:75]: We should now be able to actually remove the Link and pass all those props to the button

**Review by JulesBelveze — COMMENTED** [review]
Left a bunch of comments, let me know if you want me to pick up 👌🏼

File: sparkle/src/components/CardButton.tsx (hunk 5 above)
Comment (aubin-tchoi) [reply sparkle/src/components/CardButton.tsx:86]: Wondering why we are not doing that ⬆️

Comment (JulesBelveze) [reply sparkle/src/components/CardButton.tsx:86]: I actually tried and our Button doesn't render "children" properly (we actually should not let you do that btw)

Comment (aubin-tchoi) [reply sparkle/src/components/CardButton.tsx:86]: not super satisfying but I'll take it!

Comment (JulesBelveze) [reply sparkle/src/components/CardButton.tsx:86]: @aubin-tchoi Well we don't really want to be able to pass any kinda react node inside a Button right?

Comment (JulesBelveze) [reply sparkle/src/components/CardButton.tsx:86]: Open to alternatives though 🤓

File: sparkle/src/components/CardButton.tsx [hunk 6]
```diff
@@ -69,15 +66,21 @@ export function CardButton({
...
 }: CardButtonProps) {
   const { components } = React.useContext(SparkleContext);
   const Link: SparkleContextLinkType = href ? components.link : noHrefLink;
 
+  const cardButtonClassNames = cn(
+    cardButtonVariants({ variant, size }),
+    className
+  );
+
   if (href) {
```
Comment (aubin-tchoi) [inline sparkle/src/components/CardButton.tsx:79]: Hot take: can we remove this feature of optionally passing a href to the CardButton and limit its scope to a simple card that does smth when hovered and another thing when clicked? This component lowkey feels like two very different components at once.

Comment (aubin-tchoi) [reply sparkle/src/components/CardButton.tsx:79]: AFAICT the version of the CardButton where we pass a href is not used in front.

File: sparkle/src/components/CardButton.tsx (hunk 5 above)
Comment (aubin-tchoi) [reply sparkle/src/components/CardButton.tsx:86]: yeah sure, but I don't like that we would want to pass children here, I feel like there is something wrong with this component being called a Button

File: sparkle/src/components/CardButton.tsx (hunk 6 above)
Comment (JulesBelveze) [reply sparkle/src/components/CardButton.tsx:79]: Well not sure to follow why does it feel like two components. One could want to open a link when the card is linked right ? AFAICT it basically behaves as a regular button

File: sparkle/src/components/CardButton.tsx (hunk 5 above)
Comment (JulesBelveze) [reply sparkle/src/components/CardButton.tsx:86]: Agree on the naming, I would rather call it ClickableCard which is all it does in the end, right? 😅

Comment (JulesBelveze) [reply sparkle/src/components/CardButton.tsx:86]: Which I suspect, is the reason we use to have a div and not button. We could also rollback to it?

Comment (aubin-tchoi) [reply sparkle/src/components/CardButton.tsx:86]: Leaning toward a rollback yeah

File: sparkle/src/components/CardButton.tsx (hunk 6 above)
Comment (aubin-tchoi) [reply sparkle/src/components/CardButton.tsx:79]: just pointing out that the subsets of props that are used between the two behaviors seem a little disjoint, if I give an `onMouseEnter` and an `href` the former will be ignored. In order to share the style maybe having two components, one with a link and another that is clickable, would be more readable

**Review by aubin-tchoi — APPROVED** [review]
LGTM 🙏

**Comment by Duncid** [comment]
Thank you guys!

**Pull request merged**